app/src/main/java/com/github/polypoly/app/ui/menu/lobby/CreateGameLobbyActivity.kt

```diff
@@ -0,0 +1,334 @@
+import android.os.Bundle
+import androidx.activity.compose.setContent
+import androidx.compose.foundation.layout.*
+import androidx.compose.material.*
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.ArrowForward
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.rotate
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.dp
+import com.github.polypoly.app.base.menu.lobby.GameLobby
+import com.github.polypoly.app.base.menu.lobby.GameMode
+import com.github.polypoly.app.base.menu.lobby.GameParameters
+import com.github.polypoly.app.ui.menu.MenuActivity
+import com.github.polypoly.app.ui.theme.PolypolyTheme
+import com.github.polypoly.app.ui.theme.UIElements
+import com.github.polypoly.app.utils.Constants.Companion.GAME_LOBBY_INITIAL_BALANCE_DEFAULT
+import com.github.polypoly.app.utils.Constants.Companion.GAME_LOBBY_INITIAL_BALANCE_STEP
+import com.github.polypoly.app.utils.Constants.Companion.GAME_LOBBY_MAX_INITIAL_BALANCE
+import com.github.polypoly.app.utils.Constants.Companion.GAME_LOBBY_MAX_PLAYERS
+import com.github.polypoly.app.utils.Constants.Companion.GAME_LOBBY_MAX_ROUNDS
+import com.github.polypoly.app.utils.Constants.Companion.GAME_LOBBY_MENU_PICKER_WIDTH
+import com.github.polypoly.app.utils.Constants.Companion.GAME_LOBBY_MIN_PLAYERS
+import com.github.polypoly.app.utils.Constants.Companion.GAME_LOBBY_MIN_ROUNDS
+import com.github.polypoly.app.utils.Constants.Companion.GAME_LOBBY_PRIVATE_DEFAULT
+import com.github.polypoly.app.utils.Constants.Companion.GAME_LOBBY_ROUNDS_DEFAULT
+import com.github.polypoly.app.utils.Constants.Companion.GAME_LOBBY_ROUNDS_DURATIONS
+import com.github.polypoly.app.utils.Constants.Companion.GAME_LOBBY_ROUND_DURATION_DEFAULT
+import com.github.polypoly.app.utils.global.GlobalInstances.Companion.currentUser
+import com.github.polypoly.app.utils.global.GlobalInstances.Companion.uniqueCodeGenerator
+import com.github.polypoly.app.utils.Constants.Companion.GAME_LOBBY_MIN_INITIAL_BALANCE as GAME_LOBBY_MIN_INITIAL_BALANCE1
+
+class CreateGameLobbyActivity :  MenuActivity("Create a game") {
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        setContent {
+            MenuContent{
+                CreateGameLobbyContent()
+            }
+        }
+    }
+
+    @Composable
+    fun CreateGameLobbyContent() {
+
+        PolypolyTheme {
+            GameSettingsMenu()
+        }
+    }
+
+    /**
+     * The menu for creating a game
+     */
+    @Composable
+    fun GameSettingsMenu() {
+        val mContext = LocalContext.current
+
+        val gameName = remember { mutableStateOf("") }
```

Comment:
As you only use the `MutableState`s in this function, assigning with `by` instead of `=` would be a better idea, you wouldn't need to call `.value()` each time. If there is another reason for your choice, please tell me :)

Reply:
good idea, I juste had trouble with it so I went for '=' style. Now I found what bothered me and it works well